zingolib/src/wallet/keys/unified.rs

```diff
@@ -331,6 +298,57 @@ impl WalletCapability {
         Ok(ua)
     }
 
+    /// Generates a transparent receiver if the wallet is capable of it.
+    ///
+    /// If the wallet is not capable of generating a transparent receiver,
+    /// `None` is returned.
+    ///
+    /// The generated receiver is stored in the wallet if the wallet lock is
+    /// acquired.
+    pub fn generate_transparent_receiver(&self, desired_receivers: ReceiverSelection) -> Result<Option<Secp256k1PublicKey>, String> {
```

Comment:
non-blocking comment: I would prefer to use `secp256k1::PublicKey` I don't see the need to alias this as it gives the same information to the reader. it's your call here

Reply:
```suggestion
    pub fn generate_transparent_receiver(&self, desired_receivers: ReceiverSelection) -> Result<Option<secp256k1::PublicKey>, String> {
```